### ISSUE TITLE
Zvals are not `Send` nor `Sync`

### ISSUE THREAD

**Issue by davidcole1340** [extphprs/ext-php-rs#88]
Zvals currently implement both `Send` and `Sync`, however, if the zval contains a reference counted type (strings, arrays, objects), when dropped it will attempt to decrement the reference counter, which is not atomic and therefore not defined behaviour.

**Comment by kakserpom** [comment]
Is it not atomic in ZTS?

**Comment by ptondereau** [comment]
No, it's not atomic even in ZTS builds. ZTS only provides thread-local globals isolation, not atomic reference counting for individual values. So unfortunately, we won't fix this.
